### DISCUSSION TITLE
Understanding Widget destroy

### DISCUSSION THREAD

**Discussion by WhoAteDaCake** [lc-soft/LCUI#243] [Q&A] [answered]
Hi, this might be a stupid question, but is there a time delay on how fast you can delete widgets?

I'm attempting to build a higher level wrapper using crystal-lang. I've created a list of items, but when removing them I got Invalid memory access

```
[native-ui] Invalid memory access (signal 11) at address 0x28f
[native-ui] [0x556d1bafac96] *Exception::CallStack::print_backtrace:(Int32 | Nil) +118
[native-ui] [0x556d1bae8af9] __crystal_sigfault_handler +361
[native-ui] [0x7f9d8a3b33c0] ???
[native-ui] [0x7f9d8a5c3e57] ???
[native-ui] [0x7f9d8a5c4288] ???
[native-ui] [0x7f9d8a5c4bc5] ???
[native-ui] [0x7f9d8a5a1b29] ???
[native-ui] [0x7f9d8a5a9654] EventTrigger_Trigger +160
[native-ui] [0x7f9d8a5a1cfb] LCUI_TriggerEvent +110
[native-ui] [0x7f9d8a5b1258] ???
[native-ui] [0x7f9d8a5a9654] EventTrigger_Trigger +160
[native-ui] [0x7f9d8a5b0f18] ???
[native-ui] [0x7f9d8a5b0f7d] ???
[native-ui] [0x7f9d8a5a1ead] LCUI_ProcessEvents +44
[native-ui] [0x7f9d8a5a1a51] LCUI_RunFrame +18
[native-ui] [0x7f9d8a5a212a] LCUIMainLoop_Run +278
[native-ui] [0x7f9d8a5a281f] LCUI_Main +33
[native-ui] [0x556d1bae8cd1] *main:Int32 +113
[native-ui] [0x556d1bad9e5c] __crystal_main +1180
[native-ui] [0x556d1bb6d67d] *Crystal::main_user_code<Int32, Pointer(Pointer(UInt8))>:Nil +45
[native-ui] [0x556d1bb6d4b5] *Crystal::main<Int32, Pointer(Pointer(UInt8))>:Int32 +101
[native-ui] [0x556d1bae55ad] main +45
[native-ui] [0x7f9d8a15a0b3] __libc_start_main +243
[native-ui] [0x556d1bad98fe] _start +46
[native-ui] [0x0] ???
```
If I try to remove the first item and then the first item again, I get

> malloc(): unsorted double linked list corrupted

I'll see if I can reproduce it tomorrow, but this seem like an issue with my code rather than the library right ?
As a side node, if I'm building an abstraction on top of LCUI, should I leave unbinding children and removing event listeners to LCUI `Widget_Destroy` ? 

Thanks!

**Comment by WhoAteDaCake** [comment, marked as answer]
Turns out the reason this issue happend, was because I called unlink on a child before destroing it

**Reply by lc-soft** [reply]
This document may be helpful to you. https://docs.lcui.lc-soft.io/widget/lifecycle

> Note: This document is written in Chinese, please use a translation tool to read it.
